oneview_redfish_toolkit/app.py

```diff
@@ -100,6 +108,35 @@
     app.register_blueprint(network_port)
     app.register_blueprint(session)
 
+    @app.before_request
+    def check_authentication():
+        # If authentication_mode = conf we do nothing
+        auth_mode = util.config["redfish"]["authentication_mode"]
+        if auth_mode == "conf":
+            return None
+        else:
+            # ServiceRoot don't need auth
+            if request.path in {"/redfish/v1", "/redfish/v1/", "/redfish",
+                                "/redfish/"}:
+                g.oneview_client = util.get_oneview_client(None, True)
+                return None
+            # If authenticating also we do nothing
+            if request.path == "/redfish/v1/SessionService/Sessions" and \
+                request.method == "POST":
+                return
+            # Any other path we demand auth
+            x_auth_token = request.headers.get('x-auth-token')
+            if not x_auth_token:
+                abort(
+                    status.HTTP_401_UNAUTHORIZED,
+                    "x-auth-token header not found")
+            else:
+                try:
+                    oneview_client = util.get_oneview_client(x_auth_token)
+                except Exception:
+                    abort(status.HTTP_401_UNAUTHORIZED, "invalid auth token")
+                g.oneview_client = oneview_client
```

Comment:
It is better to put `g.oneview_client = oneview_client` on the try block, because `oneview_client` is a local variable there.

Reply:
not fixed.

Reply:
Why not? To call `oneview_client` out of its scope it's not a good thing.